File: project_euler/10_c.rs
fn primes_up_to(n: usize) -> Vec<usize> {
    let mut found_primes: Vec<usize> = Vec::new();

    'outer: for candidate in 2..=n {
        for &prime in &found_primes {
            if prime * prime > candidate {
                break;
            }
            if candidate % prime == 0 {
                continue 'outer;
            }
        }
        found_primes.push(candidate);
    }

    found_primes
}

fn main() {
    println!("{}", primes_up_to(2_000_000).iter().sum::<usize>());
}

File: project_euler/10_c.py
def primes_up_to(n: int) -> list[int]:
    # could use sieve for efficiency
    found_primes = []
    for candidate in range(2, n + 1):
        if not any(
            candidate % prime == 0
            for prime in found_primes
            if prime * prime <= candidate
        ):
            found_primes.append(candidate)

    return found_primes


print(sum(primes_up_to(2_000_000)))
